File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraReports.UI
' ...

Namespace ManageReportBands
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub AddMarginBands()

			' Check if the TopMargin band already presents in the report.
			If Bands.GetBandByType(GetType(TopMarginBand)) Is Nothing Then
				' Create a new TopMargin band and add it to the report.
				Dim tmBand As New TopMarginBand()
				Bands.Add(tmBand)

				' Create a label and set its text and width.
				Dim label As New XRLabel()
				label.Text = "TopMargin Band"
				label.Width = 200

				' Add the label to the TopMargin band.
				tmBand.Controls.Add(label)
			End If

			' Check if the BottomMargin band already presents in the report.
			If Bands.GetBandByType(GetType(BottomMarginBand)) Is Nothing Then
				' Create a new BottomMargin band and add it to the report.
				Dim bmBand As New BottomMarginBand()
				Bands.Add(bmBand)

				' Create an XRPageInfo object and set its width and PageInfo option.
				Dim datePageInfo As New XRPageInfo()
				datePageInfo.Width = 200
				datePageInfo.PageInfo = PageInfo.DateTime

				' Add the page information control to the BottomMargin band.
				bmBand.Controls.Add(datePageInfo)
			End If
		End Sub


		Public Sub RemoveMarginBands()
			' Obtain the TopMargin band and remove it from the report.
			Dim band As Band = Bands.GetBandByType(GetType(TopMarginBand))
			If band IsNot Nothing Then
				Bands.Remove(band)
			End If

			' Obtain the BottomMargin band and remove it from the report.
			band = Bands.GetBandByType(GetType(BottomMarginBand))
			If band IsNot Nothing Then
				Bands.Remove(band)
			End If
		End Sub

	End Class
End Namespace

File: CS/XtraReport1.cs
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;
// ...

namespace ManageReportBands {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        public void AddMarginBands() {

            // Check if the TopMargin band already presents in the report.
            if (Bands.GetBandByType(typeof(TopMarginBand)) == null) {
                // Create a new TopMargin band and add it to the report.
                TopMarginBand tmBand = new TopMarginBand();
                Bands.Add(tmBand);

                // Create a label and set its text and width.
                XRLabel label = new XRLabel();
                label.Text = "TopMargin Band";
                label.Width = 200;

                // Add the label to the TopMargin band.
                tmBand.Controls.Add(label);
            }

            // Check if the BottomMargin band already presents in the report.
            if (Bands.GetBandByType(typeof(BottomMarginBand)) == null) {
                // Create a new BottomMargin band and add it to the report.
                BottomMarginBand bmBand = new BottomMarginBand();
                Bands.Add(bmBand);

                // Create an XRPageInfo object and set its width and PageInfo option.
                XRPageInfo datePageInfo = new XRPageInfo();
                datePageInfo.Width = 200;
                datePageInfo.PageInfo = PageInfo.DateTime;

                // Add the page information control to the BottomMargin band.
                bmBand.Controls.Add(datePageInfo);
            }
        }


        public void RemoveMarginBands() {
            // Obtain the TopMargin band and remove it from the report.
            Band band = Bands.GetBandByType(typeof(TopMarginBand));
            if (band != null) Bands.Remove(band);

            // Obtain the BottomMargin band and remove it from the report.
            band = Bands.GetBandByType(typeof(BottomMarginBand));
            if (band != null) Bands.Remove(band);
        }

    }
}
